WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -122,5 +128,81 @@ public boolean onOptionsItemSelected(final MenuItem item) {
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putSerializable(WordPress.SITE, mSite);
+        if (!TextUtils.isEmpty(mCurrentSearch)) {
+            outState.putString(EXTRA_SEARCH_TERM, mCurrentSearch);
+        }
+    }
+
+    @Override
+    public void setTitle(CharSequence title) {
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setTitle(title);
+        } else {
+            super.setTitle(title);
+        }
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        if (mSearchAction != null) {
+            mSearchAction.collapseActionView();
+        }
+        setTitle("'" + query + "'");
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        mPostList.filterPosts(query);
+        return false;
+    }
+
+    public boolean isRefreshing() {
+        return mPostList.isRefreshing();
+    }
+
+    private PostsListFragment getPostListFragment() {
+        PostsListFragment frag = (PostsListFragment) getFragmentManager().findFragmentById(R.id.postList);
+        if (frag == null) {
+            frag = PostsListFragment.newInstance(mSite);
+        }
+        return frag;
+    }
+
+    private void unpackIntent(@NonNull Intent intent) {
+        if (intent.hasExtra(WordPress.SITE)) {
+            mSite = (SiteModel) intent.getSerializableExtra(WordPress.SITE);
+        }
+        if (intent.hasExtra(EXTRA_VIEW_PAGES)) {
+            mIsPage = intent.getBooleanExtra(EXTRA_VIEW_PAGES, false);
+        }
+        if (intent.hasExtra(EXTRA_SEARCH_TERM)) {
```

Comment:
I don't think we should be checking the `intent` for this extra, and instead should retrieve it from the `savedInstanceState` if present. As a consequence, if you rotate the device, the search term disappears.